From 72c4f148ef88b4bffb2106b99434da5c05c0bb64 Mon Sep 17 00:00:00 2001
From: Carlos Garnacho <carlosg@gnome.org>
Date: Tue, 10 Mar 2020 13:23:17 +0100
Subject: [PATCH] windowManager: Do not shutdown ibus/xsettings on X11
 compositor restart

These paths are meant for Xwayland, not for X11 compositors being restarted
through alt-f2 + r. Maybe some signal analogous to init-xserver should be
added for Xwayland shutdown paths, but this signal we are currently
listening for is backend agnostic.

Closes: https://gitlab.gnome.org/GNOME/gnome-shell/issues/2329
---
 js/ui/windowManager.js | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/js/ui/windowManager.js b/js/ui/windowManager.js
index 8f0e8943da..21c63b5c32 100644
--- a/js/ui/windowManager.js
+++ b/js/ui/windowManager.js
@@ -924,6 +924,8 @@ var WindowManager = class {
             return true;
         });
         global.display.connect('x11-display-closing', () => {
+            if (!Meta.is_wayland_compositor())
+                return;
             Shell.util_stop_systemd_unit('gsd-xsettings.target', 'fail');
             IBusManager.getIBusManager().restartDaemon();
         });
-- 
2.24.1

